src/main/java/storage/Storage.java

```diff
@@ -50,37 +42,15 @@ public class Storage {
     private Storage() {
         assert DATA_PATH.endsWith(".json") : "data file path should be of type .json";
         logger = Logger.getLogger(this.getClass().getName());
-        setupDataFile();
```

Comment:
Great job on reducing the coupling the of the data file as per our offline discussions.